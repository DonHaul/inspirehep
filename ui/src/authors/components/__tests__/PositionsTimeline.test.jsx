import React from 'react';
import { render } from '@testing-library/react';
import { fromJS } from 'immutable';

import PositionsTimeline from '../PositionsTimeline';

describe('PositionsTimeline', () => {
  it('renders with positions', () => {
    const positions = fromJS([
      {
        institution: 'Inst 1',
        display_date: '1990-1994',
        rank: 'UNDERGRADUATE',
      },
      {
        institution: 'Inst 2',
        display_date: '1994-2000',
        rank: 'PHD',
      },
      {
        institution: 'CERN',
        display_date: '2000-present',
        rank: 'STAFF',
        current: true,
      },
    ]);
    const { asFragment } = render(<PositionsTimeline positions={positions} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with single position', () => {
    const positions = fromJS([
      {
        institution: 'Inst 1',
        rank: 'UNDERGRADUATE',
        display_date: 'present',
        current: true,
      },
    ]);
    const { asFragment } = render(<PositionsTimeline positions={positions} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders only first 3 by default', () => {
    const positions = fromJS([
      { institution: 'Inst 1' },
      { institution: 'Inst 2' },
      { institution: 'Inst 3' },
      { institution: 'Inst 4' },
      { institution: 'Inst 5' },
      { institution: 'Inst 6' },
    ]);
    const { asFragment } = render(<PositionsTimeline positions={positions} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders all when ExpandListToggle is toggled by default', () => {
    const positions = fromJS([
      { institution: 'Inst 1' },
      { institution: 'Inst 2' },
      { institution: 'Inst 3' },
      { institution: 'Inst 4' },
      { institution: 'Inst 5' },
      { institution: 'Inst 6' },
    ]);
    const { getByText, getByRole } = render(
      <PositionsTimeline positions={positions} />
    );
    const toggleButton = getByRole('button');
    toggleButton.click();
    expect(getByText('Inst 4')).toBeInTheDocument();
    expect(getByText('Inst 5')).toBeInTheDocument();
    expect(getByText('Inst 6')).toBeInTheDocument();
  });
});
