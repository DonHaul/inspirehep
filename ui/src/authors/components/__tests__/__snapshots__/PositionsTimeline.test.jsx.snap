// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PositionsTimeline renders only first 3 by default 1`] = `
<DocumentFragment>
  <ul
    class="ant-timeline"
  >
    <li
      class="ant-timeline-item"
    >
      <div
        class="ant-timeline-item-tail"
      />
      <div
        class="ant-timeline-item-head ant-timeline-item-head-blue"
      />
      <div
        class="ant-timeline-item-content"
      >
        <div />
        <div>
          <span>
            Inst 1
          </span>
        </div>
      </div>
    </li>
    <li
      class="ant-timeline-item"
    >
      <div
        class="ant-timeline-item-tail"
      />
      <div
        class="ant-timeline-item-head ant-timeline-item-head-blue"
      />
      <div
        class="ant-timeline-item-content"
      >
        <div />
        <div>
          <span>
            Inst 2
          </span>
        </div>
      </div>
    </li>
    <li
      class="ant-timeline-item ant-timeline-item-last"
    >
      <div
        class="ant-timeline-item-tail"
      />
      <div
        class="ant-timeline-item-head ant-timeline-item-head-blue"
      />
      <div
        class="ant-timeline-item-content"
      >
        <div />
        <div>
          <span>
            Inst 3
          </span>
        </div>
      </div>
    </li>
  </ul>
  <button
    class="__SecondaryButton__"
    type="button"
  >
    Show all positions (6)
  </button>
</DocumentFragment>
`;

exports[`PositionsTimeline renders with positions 1`] = `
<DocumentFragment>
  <ul
    class="ant-timeline"
  >
    <li
      class="ant-timeline-item"
    >
      <div
        class="ant-timeline-item-tail"
      />
      <div
        class="ant-timeline-item-head ant-timeline-item-head-blue"
      />
      <div
        class="ant-timeline-item-content"
      >
        <div>
          1990-1994
        </div>
        <div>
          <strong>
            UNDERGRADUATE, 
          </strong>
          <span>
            Inst 1
          </span>
        </div>
      </div>
    </li>
    <li
      class="ant-timeline-item"
    >
      <div
        class="ant-timeline-item-tail"
      />
      <div
        class="ant-timeline-item-head ant-timeline-item-head-blue"
      />
      <div
        class="ant-timeline-item-content"
      >
        <div>
          1994-2000
        </div>
        <div>
          <strong>
            PHD, 
          </strong>
          <span>
            Inst 2
          </span>
        </div>
      </div>
    </li>
    <li
      class="ant-timeline-item ant-timeline-item-last"
    >
      <div
        class="ant-timeline-item-tail"
      />
      <div
        class="ant-timeline-item-head ant-timeline-item-head-blue"
      />
      <div
        class="ant-timeline-item-content"
      >
        <div>
          2000-present
        </div>
        <div>
          <strong>
            STAFF, 
          </strong>
          <span>
            CERN
          </span>
        </div>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`PositionsTimeline renders with single position 1`] = `
<DocumentFragment>
  <ul
    class="ant-timeline"
  >
    <li
      class="ant-timeline-item ant-timeline-item-last"
    >
      <div
        class="ant-timeline-item-tail"
      />
      <div
        class="ant-timeline-item-head ant-timeline-item-head-blue"
      />
      <div
        class="ant-timeline-item-content"
      >
        <div>
          present
        </div>
        <div>
          <strong>
            UNDERGRADUATE, 
          </strong>
          <span>
            Inst 1
          </span>
        </div>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
